sematic/api/server.py

```diff
@@ -161,18 +148,37 @@ def handler(signum, frame):
     signal.signal(signal.SIGHUP, handler)
 
 
-def run_socketio(debug=False):
-    with open(get_config().server_pid_file_path, "w+") as fp:
-        fp.write(str(os.getpid()))
+def init_socket_io_server(make_async):
+    server_class = socketio.AsyncServer if make_async else socketio.Server
+    async_mode = "asgi" if make_async else "threading"
+    namespace_class = AsyncNamespace if make_async else Namespace
 
-    dictConfig(make_log_config(log_to_disk=True))
+    sio = server_class(async_mode=async_mode, cors_allowed_origins="*")
+    for namespace in SOCKET_IO_NAMESPACES:
+        sio.register_namespace(namespace_class(namespace))
+    register_sio_server(sio)
+    return sio
+
+
+def run_locally(debug=False, make_daemon=False):
+    dictConfig(make_log_config(log_to_disk=True, level=logging.INFO))
     register_signal_handlers()
 
-    socketio.run(
+    sio = init_socket_io_server(make_async=False)
+    sematic_api.wsgi_app = socketio.WSGIApp(sio, sematic_api.wsgi_app)
+    if make_daemon:
+        daemonize(False)
+
+    with open(get_config().server_pid_file_path, "w+") as fp:
+        fp.write(str(os.getpid()))
+
+    uvicorn.run(
```

Comment:
How can the number of threads be specified?

Reply:
I don't think there's any setting relating to number of threads, but the number of worker processes must be 1 when using `uvicorn.run`. This appears to be enough for local usage.